02-array-shift/lib/array_shift.js

```diff
@@ -0,0 +1,17 @@
+'use strict';
+// module.exports = exports = {};
+module.exports = exports = {};
+
+exports.insertShiftArray = (inputArr, item) => {
+  const length = inputArr.length
+  const middle = Math.ceil(inputArr.length/2 -1);
+  let temp = inputArr[middle]; 
+  inputArr[middle] = item;
+  
+  for (let i = middle; i <= length; i++){
+    let tempTwo = inputArr[i];
+    inputArr[i] = temp;
+    temp = tempTwo;
```

Comment:
If you're using a lot of temporary variables such as `temp`, it might be a sign that the logic is not as optimal as it could be. More than one `temp` can also reduce readability. Overall, it's a pretty good solution though.